### PR TITLE
Ensure we set '00' hesa_disabilities code

### PR DESCRIPTION
**Pull request by steventux**
## Context

If the candidate has already answered some of the equality and diversity questions but answers 'no' to disability status we still need to populate the appropriate `00` HESA code.
The code was previously expecting the equality_and_diversity field to be nil.

Similarly if a candidate changes their answer to 'no' for an application
with existing disabilities data we should set the appropriate '00' code.

## Changes proposed in this pull request

Ensures we set the `00` no disabilities HESA code when there is existing equality and diversity data (ie when the candidate has already answered other parts of the questionnaire).
 

## Link to Trello card

https://trello.com/c/cQurakCV/3772-tribal-select-no-disability-returns-blank-should-be-00

## Things to check

- [x] This code does not rely on migrations in the same Pull Request
- [x] If this code includes a migration adding or changing columns, it also backfills existing records for consistency
- [x] API release notes have been updated if necessary
- [x] Required environment variables have been updated [added to the Azure KeyVault](/docs/environment-variables.md#deploy-pipeline)

### CODE REVIEW

File: app/forms/candidate_interface/equality_and_diversity/disability_status_form.rb [hunk 1]
```diff
@@ -18,18 +18,15 @@ def self.build_from_application(application_form)
     def save(application_form)
       return false unless valid?
 
-      if application_form.equality_and_diversity.nil?
-        application_form.update(equality_and_diversity: {
-          'disabilities' => [],
-          'hesa_disabilities' => [hesa_code].compact,
-        })
-      elsif reset_disabilities?(application_form)
+      application_form.equality_and_diversity = {} if application_form.equality_and_diversity.nil?
```
Comment (despo) [inline app/forms/candidate_interface/equality_and_diversity/disability_status_form.rb:21]:
```suggestion
      application_form.equality_and_diversity ||= {} 
```

File: spec/forms/candidate_interface/equality_and_diversity/disability_status_form_spec.rb [hunk 2]
```diff
@@ -72,13 +72,25 @@
...
+      end
+
       it 'updates the existing record of equality and diversity information' do
         application_form = build(:application_form, equality_and_diversity: { 'sex' => 'male' })
         form = CandidateInterface::EqualityAndDiversity::DisabilityStatusForm.new(disability_status: 'yes')
         form.save(application_form)
 
         expect(application_form.equality_and_diversity).to eq(
-          'sex' => 'male', 'disabilities' => [],
+          'sex' => 'male', 'disabilities' => [], 'hesa_disabilities' => [],
```
Comment (despo) [inline spec/forms/candidate_interface/equality_and_diversity/disability_status_form_spec.rb:93]:
```suggestion
          'sex' => 'male', 'disabilities' => [], 
          'hesa_disabilities' => [],
```

File: spec/forms/candidate_interface/equality_and_diversity/disability_status_form_spec.rb [hunk 3]
```diff
@@ -98,7 +110,7 @@
         form.save(application_form)
 
         expect(application_form.equality_and_diversity).to eq(
-          'sex' => 'male', 'disabilities' => [],
+          'sex' => 'male', 'disabilities' => [], 'hesa_disabilities' => %w[00],
```
Comment (despo) [inline spec/forms/candidate_interface/equality_and_diversity/disability_status_form_spec.rb:113]:
```suggestion
          'sex' => 'male', 'disabilities' => [], 
          'hesa_disabilities' => %w[00],
```

File: spec/forms/candidate_interface/equality_and_diversity/disability_status_form_spec.rb [hunk 4]
```diff
@@ -120,7 +132,7 @@
         form.save(application_form)
 
         expect(application_form.equality_and_diversity).to eq(
-          'sex' => 'male', 'disabilities' => [],
+          'sex' => 'male', 'disabilities' => [], 'hesa_disabilities' => [],
```
Comment (despo) [inline spec/forms/candidate_interface/equality_and_diversity/disability_status_form_spec.rb:135]:
```suggestion
          'sex' => 'male', 'disabilities' => [], 
          'hesa_disabilities' => [],
```

File: app/forms/candidate_interface/equality_and_diversity/disability_status_form.rb [hunk 5]
```diff
@@ -18,18 +18,15 @@ def self.build_from_application(application_form)
...
+
+      if application_form.equality_and_diversity['disabilities'].nil? || reset_disabilities?(application_form)
```
Comment (despo) [inline app/forms/candidate_interface/equality_and_diversity/disability_status_form.rb:23]: Consider moving `application_form.equality_and_diversity['disabilities'].nil?` to `reset_disabilities?`

Comment (despo) [reply app/forms/candidate_interface/equality_and_diversity/disability_status_form.rb:23]: I just noticed it was actually moved out of there.. I don't know if I'm missing some context here, but is there any reason as to why?

**Review by despo — APPROVED** [review]
Looks good, left some minor feedback

File: app/forms/candidate_interface/equality_and_diversity/disability_status_form.rb (hunk 5 above)
Comment (steventux) [reply app/forms/candidate_interface/equality_and_diversity/disability_status_form.rb:23]: Yes, and I did wonder if having one method to answer the conditional would be better but it seemed like there are 2 different cases for how enter this condition:

1. disabilities key is missing from equality and diversity data (when the user has never filled it in)
2. disabilities key is present but the values are being 'reset' (when the user is changing their answers).

Perhaps I should have been bolder and removed the `reset_...` method naming as it's just updating? The tests seem to mention resetting a lot but I wonder if it's the right term?

Comment (despo) [reply app/forms/candidate_interface/equality_and_diversity/disability_status_form.rb:23]: Ok, I see what you mean.

**Review by cwrw — APPROVED** [review]

**Review by stevehook — APPROVED** [review]
All makes sense to me. Gave it a spin locally and seems to set the correct HESA codes. :+1:

**Pull request merged**